auth/api/iam/types.go

```diff
@@ -0,0 +1,152 @@
+//
+//	If an authorization request is missing the "response_type" parameter,
+//	or if the response type is not understood, the authorization server
+//	MUST return an error response as described in Section 4.1.2.1.
+type responseType = string
+
+const (
+	// responseTypeCode is the default response_type in the OAuth2 authorized code flow
+	responseTypeCode responseType = "code"
+	// responseTypeVPToken is defined in the OpenID4VP vp_token flow
+	// https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#appendix-B
+	responseTypeVPToken responseType = "vp_token"
+	// responseTypeVPIDToken is defined in the OpenID4VP flow that combines its vp_token with SIOPv2's id_token
+	// https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#appendix-B
+	responseTypeVPIDToken responseType = "vp_token id_token"
+)
+
+var responseTypesSupported = []responseType{responseTypeCode, responseTypeVPToken, responseTypeVPIDToken}
+
+// responseMode
+type responseMode = string
+
+const (
+	// responseModeQuery returns the answer to the authorization request append as query parameters to the provided redirect_uri
+	responseModeQuery responseMode = "query" // default if no response_mode is specified
+	// responseModeDirectPost signals the Authorization Server to POST the requested presentation definition to the provided response_uri
+	// https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-response-mode-direct_post
+	responseModeDirectPost responseMode = "direct_post"
+)
+
+var responseModesSupported = []responseMode{responseModeQuery, responseModeDirectPost}
+
+// grantType
+type grantType = string
+
+const (
+	// grantTypeAuthorizationCode is the default OAuth2 grant type
+	grantTypeAuthorizationCode grantType = "authorization_code"
+	// grantTypeVPToken is used to present a vp_token in exchange for an access_token in service-to-service flows
+	// TODO: EBSI inspired flow that is not standardized
+	// 		 https://api-conformance.ebsi.eu/docs/ct/verifiable-presentation-exchange-guidelines-v3#service-to-service-token-flow
+	grantTypeVPToken grantType = "vp_token"
+	// grantTypePreAuthorizedCode is defined in the pre-authorized_code flow of OpenID4VCI
+	// https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-sub-namespace-registration
+	grantTypePreAuthorizedCode grantType = "urn:ietf:params:oauth:grant-type:pre-authorized_code"
+)
+
+var grantTypesSupported = []grantType{grantTypeAuthorizationCode, grantTypeVPToken, grantTypePreAuthorizedCode}
+
+// algValuesSupported contains a list of supported cipher suites for jwt_vc_json & jwt_vp_json presentation formats
+// Recommended list of options https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms
+// TODO: validate list, should reflect current recommendations from https://www.ncsc.nl
+var algValuesSupported = []string{"PS256", "PS384", "PS512", "ES256", "ES384", "ES512"}
```

Comment:
check with ARF

Reply:
I can't find the exact details on this. I might be searching for the wrong terms

Reply:
Then it's probably not in there.